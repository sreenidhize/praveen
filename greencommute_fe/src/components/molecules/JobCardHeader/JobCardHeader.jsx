import React from "react";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";
import CompanyLogo from "../../atom/CompanyLogo/CompanyLogo";
import JobCardLabel from "../../atom/JobCardLabel/JobCardLabel";
import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
  buttonsStyle: {
    width: "220px",
    marginTop: "12px",
  },

  columnMargin: {
    marginLeft: "20px",
  },

  root: {
    display: "flex",
  },
  smallMedia: {
    width: 50,
    height: 50,
  },

  primary: {
    fontSize: "16px",
    color: "#19293b",
    marginBottom: "6px",
    width: "17rem",
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
  secondary: {
    fontSize: "12px",
    color: "#5f7381",
    marginBottom: "6px",
    width: "17rem",
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis"
  },
});

const JobCardHeader = ({ src, alt, jobTitle, companyName, location }) => {
  const classes = useStyles();

  return (
    <Grid className={classes.root}>
      <Grid item>
        <CompanyLogo
          source={src}
          alt={alt}
          {...{ className: classes.smallMedia }}
        ></CompanyLogo>
      </Grid>

      <Grid container direction="column" className={classes.columnMargin}>
        <Grid item>
          <JobCardLabel label={jobTitle} {...{ className: classes.primary }} />
        </Grid>
        <Grid item>
          <JobCardLabel
            label={companyName}
            {...{ className: classes.secondary }}
          >
            {" "}
          </JobCardLabel>
        </Grid>
        <Grid item>
          <JobCardLabel label={location} {...{ className: classes.secondary }}>
            {" "}
          </JobCardLabel>
        </Grid>
      </Grid>
    </Grid>
  );
};

JobCardHeader.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  jobTitle: PropTypes.string,
  companyName: PropTypes.string,
  location: PropTypes.string,
};

export default JobCardHeader;
