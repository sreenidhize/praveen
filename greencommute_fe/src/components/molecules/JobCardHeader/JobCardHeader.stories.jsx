import React from "react";
import JobCardHeader from "./JobCardHeader";
export default {
  title: "Molecules/CardHeader",
  component: JobCardHeader,
};
const JobCardHeaderTemplate = (args) => <JobCardHeader {...args} />;

export const Primary = JobCardHeaderTemplate.bind({});

Primary.args = {
  src: "https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg",
  alt: "BMW company",
  jobTitle: "User Experience Designer",
  companyName: "BMW",
  location: "Hi-Tech City, Telangana",
};
