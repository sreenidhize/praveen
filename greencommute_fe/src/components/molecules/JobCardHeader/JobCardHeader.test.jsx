import React from "react";
import JobCardHeader from "./JobCardHeader";
import { render } from "@testing-library/react";

describe("JobCardHeader renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<JobCardHeader />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("JobCardHeader should not be empty", () => {
    const { container } = render(
      <JobCardHeader
        src="https://upload.wikimedia.org/wikipedia/commons/4/44/BMW.svg"
        alt="BMW company"
        jobTitle="User Experience Designer"
        companyName="BMW"
        location="Hi-Tech City, Telangana"
      />
    );
    expect(container.getElementsByTagName("img").length).toBe(1);
    expect(container.getElementsByTagName("h4").length).toBe(3);
  });
});
