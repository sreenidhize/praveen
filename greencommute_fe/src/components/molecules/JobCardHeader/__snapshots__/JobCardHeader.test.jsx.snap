// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCardHeader renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-root-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-smallMedia-4 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-columnMargin-2 MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h4
          class="MuiTypography-root makeStyles-primary-5 MuiTypography-h4"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h4
          class="MuiTypography-root makeStyles-secondary-6 MuiTypography-h4"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h4
          class="MuiTypography-root makeStyles-secondary-6 MuiTypography-h4"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
